pbc/bragg_peak.py

```diff
@@ -1,36 +1,104 @@
 import logging
 
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import interpolate
+
+logging.basicConfig(level=0)
 logger = logging.getLogger(__name__)
 
 
-class BraggPeak:
-    def __init__(self, bragg_peak, primary_position, shifted_position, height):
-        # todo: merge those?
-        self.peak = bragg_peak
-        self.primary_pos = primary_position
-        self.shifted_pos = shifted_position
-        self.height = height
-        self.bp_data = {"bp": self.peak,
-                        "properties":
-                            {"primary_position": self.primary_pos,
-                             "shifted_position": self.shifted_pos,
-                             "height": self.height
-                             }
-                        }
-        logger.debug("Creating BraggPeak - primary position %.2f, shifted position %.2f height %.2f\nfrom\n%s" % (
-            primary_position, shifted_position, height, bragg_peak))
+class BraggPeak(object):
+    def __init__(self, bp_domain, bp_vals):
+        if len(bp_domain) != len(bp_vals):
```

Comment:
No, they are used for `self.spline = interpolate.InterpolatedUnivariateSpline(bp_domain, bp_vals, ext=3) ` and dropped. Should I save them?